src/mpcCoreKit.ts

```diff
@@ -807,13 +814,29 @@ export class Web3AuthMPCCoreKit implements ICoreKit, IMPCContext {
     };
   }
 
-  public async sign(data: Buffer, hashed: boolean = false, secp256k1Precompute?: Secp256k1PrecomputedClient): Promise<Buffer> {
+  public async sign(
+    data: Buffer,
+    opts?: {
+      hashed?: boolean;
+      secp256k1Precompute?: Secp256k1PrecomputedClient;
+      keyTweak?: BN;
+    }
+  ): Promise<Buffer> {
     this.wasmLib = await this.loadTssWasm();
     if (this._sigType === "ecdsa-secp256k1") {
-      const sig = await this.sign_ECDSA_secp256k1(data, hashed, secp256k1Precompute);
+      if (opts?.keyTweak) {
+        throw CoreKitError.default("key tweaking not supported for ecdsa-secp256k1");
+      }
+      const sig = await this.sign_ECDSA_secp256k1(data, opts?.hashed, opts?.secp256k1Precompute);
       return Buffer.concat([sig.r, sig.s, Buffer.from([sig.v])]);
     } else if (this._sigType === "ed25519" || this._sigType === "bip340") {
-      return this.sign_frost(data, hashed);
+      if (opts?.hashed) {
+        throw CoreKitError.default(`hashed data not supported for bip340`);
```

Comment:
i think at this point we should streamline this hashed parameter, having it supported for one sig type and not for others increases integration  and doc effort. Either we should drop the support of it completely or should make it consistent.

Happy to create a diff issue for that.